### PR TITLE
feat: update slippage tolerance to use auto or custom

### PR DESCRIPTION
**Pull request by ianlapham**
- change slippage type to `Percent | auto` to support auto slippage calculations based on trade details 

NB: This is a functional change - design/UI updates will be made in a separate PR.

### CODE REVIEW

File: src/state/swap/hooks.tsx
Comment (zzmp) [inline src/state/swap/hooks.tsx:137]: Neither of these function names refer to auto slippage. Is "auto slippage" === "slippage tolerance from trade"? Can one or the other be re-named?

File: src/lib/state/settings.ts
Comment (zzmp) [inline src/lib/state/settings.ts:3]: I believe you can delete all of the `Customizable` code at this point.

Comment (zzmp) [reply src/lib/state/settings.ts:3]: If there's something using it that is not in the UI (tech debt) you can put this off to another PR / I can get rid of it after this merges.

File: src/hooks/useSlippageToleranceFromTrade.ts
Comment (zzmp) [inline src/hooks/useSlippageToleranceFromTrade.ts:33]: Type the return value to make it more obvious (and catch issues if you return the wrong thing).
Feel free to just write "Returns ..." instead of using the jsdoc keywords - this is TypeScript, so that information is captured via types anyways.

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:91]: full sentence to make it clearer

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:95]: make this logic a one-liner ternary: `() => userMaxSlippage === 'auto' ? slippageToleranceFromTrade : userMaxSlippage`

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:93]: Just call this `maxSlippage` to match the atom, not `userMaxSlippage`

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx
Comment (zzmp) [inline src/lib/components/Swap/Settings/MaxSlippageSelect.tsx:1]: These changes aren't spec'ed out (@ppphhhiiilll), and you've changed the UI so that the button styles are now different. Could you back out all changes to the UI of this file, and just change the functional aspect?

I think we need to get a spec before we add the warnings that you've added here.

**Review by zzmp — CHANGES_REQUESTED** [review]
Back out MaxSlippageSelect.tsx changes - breaks existing UI.

File: src/state/swap/hooks.tsx
Comment (ianlapham) [reply src/state/swap/hooks.tsx:137]: correct about "auto slippage" === "slippage tolerance from trade"

I like calling it autoSlippageTolerance - will rename that hook to match too

File: src/hooks/useAutoSlippageTolerance.ts
Comment (zzmp) [inline src/hooks/useAutoSlippageTolerance.ts:32]: nit: Captialize and end with a period. Makes it cleaner and more legible.

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:91]: `/**` is for doc comments. This should use `/*`.
Also, multi-line comments should start with ` * ` - usually your formatter will do this for you.

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:95]: typo: `auot`->`auto`

File: src/lib/components/Swap/Settings/MaxSlippageSelect.tsx
Comment (zzmp) [inline src/lib/components/Swap/Settings/MaxSlippageSelect.tsx:1]: You've still changed the UI here. You shouldn't be changing the `<Option>` element here except to update it to say `Auto`. Please back most of this out.

In particular, you no longer display the checkmark when the option is selected. This is a UI regression.

**Review by zzmp — CHANGES_REQUESTED** [review]
Still needs UI changes backed out.

**Review by JFrankfurt — APPROVED** [review]

File: src/lib/hooks/swap/useSwapInfo.tsx
Comment (zzmp) [inline src/lib/hooks/swap/useSwapInfo.tsx:166]: Doc comment should begin with `/**/

**Review by zzmp — APPROVED** [review]
Doc comment needs to use `/**` (as noted) - otherwise lgtm!

**Pull request merged**
